### PR TITLE
Download the effective TLD list when needed

### LINKED ISSUE #2420: remove effective_names_tld.dat from git
It is a verbatim copy of a file provided by Mozilla. It should not be in git. However, in the past we've made people unhappy by requiring Internet access during various build steps, so we should make sure we don't offend anyone by trying to fetch this file during `dist-recursor`

### PR DESCRIPTION
**Pull request by pieterlexis**
For in-tree builds, we grab this file during make. When
build-scripts/dist-recursor is run, we always grab a fresh copy.

Closes #2420

### CODE REVIEW

File: configure.ac [hunk 1]
```diff
@@ -109,6 +109,9 @@ AC_CHECK_HEADERS(
 
 PDNS_CHECK_READLINE
 PDNS_CHECK_RAGEL
+# We need curl to retrieve the effective tld list
+PDNS_CHECK_CURL_PROGRAM
```
Comment (Habbie) [inline configure.ac]: Will this make auth tarball builds depend on curl?

Comment (pieterlexis) [inline configure.ac]: yes, I was considering having a `--with-pdns-recursor` flag, but decided most systems have curl installed anyway

**Comment by cmouse** [comment]
we can't use yahttp for this?

File: configure.ac (hunk 1 above)
Comment (Habbie) [inline configure.ac]: I'd rather see `make dist` just die without `curl`, than forcing this dep on all auth users.

**Comment by Habbie** [comment]
only if we write a curl-like tool around yahttp :)

**Pull request merged**